src/System.Windows.Forms.Design/src/System/ComponentModel/Design/ToolStripContainerActionList.cs

```diff
@@ -0,0 +1,339 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.ComponentModel;
+using System.ComponentModel.Design;
+using System.Windows.Forms.Design.Behavior;
+
+namespace System.Windows.Forms.Design;
+
+/// <summary>
+///  Describes the list of actions that can be performed for the ToolStripContainer control from the chrome panel.
+/// </summary>
+internal class ToolStripContainerActionList : DesignerActionList
+{
+    private readonly ToolStripContainer _toolStripContainer;
+    private readonly IDesignerHost? _designerHost;
+    private readonly IServiceProvider? _serviceProvider;
+
+    /// <summary>
+    ///  ToolStripContainer ActionList.
+    /// </summary>
+    public ToolStripContainerActionList(ToolStripContainer toolStripContainer) : base(toolStripContainer)
+    {
+        _toolStripContainer = toolStripContainer;
+        _serviceProvider = _toolStripContainer.Site;
+        _designerHost = _serviceProvider?.GetService<IDesignerHost>();
+    }
+
+    // Helper function to get the property on the component.
+    private static object? GetProperty(Component component, string propertyName)
+    {
+        PropertyDescriptor? getProperty = TypeDescriptor.GetProperties(component)?[propertyName];
+        return getProperty?.GetValue(component);
+    }
+
+    // Helper function to change the property on the component
+    private void ChangeProperty(Component component, string propertyName, object value)
+    {
+        if (_designerHost is not null)
+        {
+            ToolStripPanel? panel = component as ToolStripPanel;
+            ToolStripPanelDesigner? panelDesigner = _designerHost.GetDesigner(component) as ToolStripPanelDesigner;
+
+            if (propertyName.Equals("Visible") && panel is not null)
+            {
+                foreach (Control control in panel.Controls)
+                {
+                    PropertyDescriptor? visibleProperty = TypeDescriptor.GetProperties(control)["Visible"];
+                    visibleProperty?.SetValue(control, value);
+                }
+
+                if (!(bool)value)
+                {
+                    if (panel is not null)
+                    {
+                        panel.Padding = new Padding(0);
+                    }
+
+                    if (panelDesigner is not null)
+                    {
+                        if (panelDesigner.ToolStripPanelSelectorGlyph is not null)
```

Comment:
Please combine the if statements
```suggestion
                    if (panelDesigner?.ToolStripPanelSelectorGlyph  is not null)
                    {
```